### PR TITLE
Miscellaneous build process improvements.

### PR DESCRIPTION
**Pull request by Kangz**
Some changes I found while sorting old branches. PTAL

### CODE REVIEW

File: libs/libRocket/Source/Core/precompiled.cpp [hunk 1]
```diff
@@ -27,3 +27,6 @@
 
 #include "precompiled.h"
 
+// XCode's ranlib complain on empty object, adding any function prevents this
+void RocketDummyFunction() {
```
Comment (DolceTriade) [inline libs/libRocket/Source/Core/precompiled.cpp]: Please make this commit in the libRocket repo. I will make this a submodule very soon.

Comment (Kangz) [inline libs/libRocket/Source/Core/precompiled.cpp]: Great :D
Let's keep this PR open until this happens.

Comment (mbasaglia) [inline libs/libRocket/Source/Core/precompiled.cpp]: If it isn't being used elsewhere might be better to declare it as static.

Comment (Kangz) [inline libs/libRocket/Source/Core/precompiled.cpp]: Yeah, static doesn't work.

Comment (DolceTriade) [inline libs/libRocket/Source/Core/precompiled.cpp]: Either way, the commit needs to happen in the librocket repo so it doesn't get lost. then we update the librocket here by copying what's in the librocket repo.

Comment (mbasaglia) [inline libs/libRocket/Source/Core/precompiled.cpp]: Out of curiosity, do we keep librocket up to date manually or do we use git subtrees?

Comment (DolceTriade) [inline libs/libRocket/Source/Core/precompiled.cpp]: Manually. It will be a submodule soon.

File: CMakeLists.txt [hunk 2]
```diff
@@ -27,6 +27,13 @@
 cmake_minimum_required(VERSION 2.8)
 project(Unvanquished C CXX)
 
+# When we build in the source dir, the daemon executable cannot be create
+# as there is already a daemon directory, so we disallow building in the
+# source dir.
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
```
Comment (mbasaglia) [inline CMakeLists.txt]: Do you think it's worth having a smarter comparison which accounts for symlinks and whatnot?

Comment (Kangz) [inline CMakeLists.txt]: I don't think so, as it would be odd for someone to have build point to its parent directory.

**Comment by mbasaglia** [comment]
Now we have the librocket submodule, so this can continue.

**Comment by Viech** [comment]
@Kangz, you having an eye on this?

**Comment by Kangz** [comment]
Now I do, added to my TODO list

**Pull request closed without merging**